core/src/test/scala/org/apache/spark/JobCancellationSuite.scala

```diff
@@ -603,7 +603,8 @@ class JobCancellationSuite extends SparkFunSuite with Matchers with BeforeAndAft
     // Kill one of the action.
     Future {
       sem1.acquire()
-      f1.cancel()
+      val reason = "custom cancel reason"
+      f1.cancel(Option(reason))
```

Comment:
Where do we verify the reason? If we don't need to verify, then maybe revert the change here?

Reply:
Sounds good, I reverted the change for this test and added another short test that verifies the reason for `FutureAction.cancel()`